src/go/k8s/pkg/resources/configmap.go

```diff
@@ -0,0 +1,171 @@
+package resources
+
+import (
+	"context"
+	"path/filepath"
+	"strconv"
+
+	redpandav1alpha1 "github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	baseSuffix	= "-base"
+	dataDirectory	= "/var/lib/redpanda/data"
+	fsGroup		= 101
+
+	configDir		= "/etc/redpanda"
+	configuratorDir		= "/mnt/operator"
+	configuratorScript	= "configurator.sh"
+)
+
+var (
+	configPath		= filepath.Join(configDir, "redpanda.yaml")
+	configuratorPath	= filepath.Join(configuratorDir, configuratorScript)
+)
+
+var _ Resource = &ConfigMapResource{}
+
+type ConfigMapResource struct {
+	client.Client
+	scheme		*runtime.Scheme
+	pandaCluster	*redpandav1alpha1.Cluster
+}
+
+func NewConfigMap(
+	client client.Client,
+	pandaCluster *redpandav1alpha1.Cluster,
+	scheme *runtime.Scheme,
+) *ConfigMapResource {
+	return &ConfigMapResource{
+		client, scheme, pandaCluster,
+	}
+}
+
+func (r *ConfigMapResource) Ensure(ctx context.Context) error {
+	var cfgm corev1.ConfigMap
+
+	err := r.Get(ctx, r.Key(), &cfgm)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if errors.IsNotFound(err) {
+		obj, err := r.Obj()
+		if err != nil {
+			return err
+		}
+		return r.Create(ctx, obj)
+	}
+	return nil
+}
+
+func (r *ConfigMapResource) Obj() (client.Object, error) {
+	serviceAddress := r.pandaCluster.Name + "." + r.pandaCluster.Namespace + ".svc.cluster.local"
```

Comment:
Not related to this PR: In future commits, perhaps we could create a function under the service resource that returns this address.